src/main/java/net/minecraftforge/fml/common/asm/transformers/BlamingTransformer.java

```diff
@@ -0,0 +1,117 @@
+package net.minecraftforge.fml.common.asm.transformers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+import net.minecraft.launchwrapper.IClassTransformer;
+import net.minecraftforge.fml.common.FMLLog;
+
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Opcodes;
+
+import com.google.common.collect.ImmutableSet;
+
+public class BlamingTransformer implements IClassTransformer
+{
+    private static final Map<String, String> classMap = new HashMap<String, String>();
+    private static final Set<String> naughtyMods = new HashSet<String>();
+    private static final Set<String> naughtyClasses = new TreeSet<String>();
+    private static final Set<String> orphanNaughtyClasses = new HashSet();
+
+    @Override
+    public byte[] transform(String name, String transformedName, byte[] bytes)
+    {
+        if (bytes == null) { return null; }
+
+        ClassReader classReader = new ClassReader(bytes);
+        ClassWriter writer = new ClassWriter(0);
+        VersionVisitor visitor = new VersionVisitor(writer);
+        classReader.accept(visitor, 0);
+        return writer.toByteArray();
```

Comment:
This can be cleaned up as we're not actually transforming, by not using a writer. And not passing in the VISIT_CODE flag to the reader.